### DISCUSSION TITLE
Change Request: allow `allowReserved` regardless of version

### DISCUSSION THREAD

**Discussion by jsumners-nr** [eslint/eslint#19256] [Q&A]
### ESLint version

9.17.0

### What problem do you want to solve?

Per this difficult to find reference https://eslint.org/docs/latest/use/configure/language-options#specifying-parser-options, the `allowReserved` option is tied to the ECMA v3 setting. I very much have a case where I need to test that reserved words do not interfere with operations. As a result, I seem to be completely blocked on being able to upgrade to eslint v9.

https://github.com/newrelic/node-newrelic/blob/47b8398820d665a85a96ae84e30eaaf20564dcf8/test/unit/instrumentation/connect.test.js#L130-L140

### What do you think is the correct solution?

Allow us to turn off this not very helpful error.

### Participation

- [ ] I am willing to submit a pull request for this change.

**Comment by aladdin-add** [comment]
it's not a blocker to upgrade to eslint v9. the change is the flat config default to `sourceType: "module"`,  so the default is strict mode .

to restore it, you can config:

```js
{languageOptions: {sourceType: "commonjs"}}
```


so it won't report: https://eslint.org/play/#eyJ0ZXh0IjoiZnVuY3Rpb24gc3RhdGljKCl7fSIsIm9wdGlvbnMiOnsicnVsZXMiOnt9LCJsYW5ndWFnZU9wdGlvbnMiOnsic291cmNlVHlwZSI6ImNvbW1vbmpzIiwicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9fX19fQ==

**Reply by jsumners-nr** [reply]
Huh?

**Comment by jsumners-nr** [comment]
Really uncool that this was converted to a discussion without any heads up. GitHub does not carry over notifications and does not send any notification that it was changed. Also, I don't see this as a discussion item.

**Reply by aladdin-add** [reply]
`static` is reversed in strict mode (it's the default in esm). The code you provided won't work properly, so I don't think eslint can allow it.

I converted it to a issue: https://github.com/eslint/eslint/issues/19278

**Reply by aladdin-add** [reply]
re the notifications - By default, you should have received the notification to the closing issue, not sure why you didn't, it might have something to do with your settings.
